src/declarativedbusinterface.cpp

```diff
@@ -351,7 +352,7 @@ void DeclarativeDBusInterface::typedCall(const QString &method, const QJSValue &
     QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(pendingCall);
     connect(watcher, SIGNAL(finished(QDBusPendingCallWatcher*)),
             this, SLOT(pendingCallFinished(QDBusPendingCallWatcher*)));
-    m_pendingCalls.insert(watcher, callback);
+    m_pendingCalls.insert(watcher, qMakePair(callback, errorCallback));
```

Comment:
I would add a check for  errorCallback.isCallable() if !errorCallback.isUndefined(). Similar to line 337.